### ISSUE TITLE
feat(policies): MotionBricksPolicy for generative G1 locomotion

### ISSUE THREAD

**Issue by cagataycali** [strands-labs/robots#467]
## Summary

Build a new `MotionBricksPolicy` provider from scratch wrapping the NVlabs
MotionBricks latent generative pipeline for real-time, demo-grade expressive
full-body G1 motion (29-DOF qpos). Clean-room implementation against the upstream
interactive demo - no prior branch is reused.

Upstream: https://github.com/NVlabs/GR00T-WholeBodyControl/tree/main/motionbricks
Reference: `motionbricks/scripts/interactive_demo_g1.py`
Project page: https://nvlabs.github.io/motionbricks/

## Motivation

WBC (issue A) gives deploy-grade stability via ONNX torque control; MotionBricks
gives stylistic, demo-grade expression (zombie/dance/stealth/crawl/...) by
generating full 29-DOF joint targets directly. Different DOF coverage, different
goal - they are complementary and, with groot upper-body manipulation, motivate a
`CompositePolicy` (issue C).

## Upstream architecture (from the package + demo)

Package `motionbricks/motionbricks/`:
`vqvae/` (motion tokenizer), `motion_backbone/` (pose + root models),
`motionlib/`, `geometry/`, `helper/`, `data/`, `exp_setup/`.

Three pretrained checkpoints (Git LFS, ~2.2 GB total, fetched from HF
`nvidia/MotionBricks-G1`): motion tokenizer / VQVAE (~273 MB), pose model
(~1.6 GB), root trajectory model (~391 MB). PyTorch required, CUDA GPU; runs at
very high throughput (state-only, no images).

Reference inference loop (`interactive_demo_g1.py`):
- builds a "full agent" + "controller" from model configs; humanoid asset is
  `assets/skeletons/g1/scene_29dof.xml`.
- per tick: `qpos = full_agent.get_next_frame()` returns a **29-DOF MuJoCo qpos**
  (plus root) written to `mj_data.qpos[:]`.
- a `control_signals` dict drives the controller; optional
  `control_signals['context_mujoco_qpos']` feeds back the achieved pose.
- 15 styles (verified upstream gallery): `idle`, `slow_walk`, `walk`,
  `hand_crawling`, `walk_boxing`, `elbow_crawling`, `stealth_walk`,
  `injured_walk`, `walk_stealth`, `walk_happy_dance`, `walk_zombie`, `walk_gun`,
  `walk_scared`, `walk_left`, `walk_right`.

## Implementation spec

New package `strands_robots/policies/motionbricks/`:

```
motionbricks/
├── __init__.py          # exports MotionBricksPolicy + JOINT_NAMES (thin)
├── policy.py            # MotionBricksPolicy(Policy)
├── pipeline.py          # thin wrapper over the 3-model agent (load + step)
├── styles.py            # MotionStyle enum (the 15 styles) + validation
└── checkpoint.py        # resolve local | HF download | cache (3 sub-models)
```

`MotionBricksPolicy(Policy)` contract:
- `__init__(self, checkpoint: str = "nvidia/MotionBricks-G1",
   style: str = "walk", allow_missing_models: bool = False,
   idle_velocity_threshold: float = 0.1)`.
- Resolves + loads all three sub-models in `__init__`; raises `RuntimeError` on
  any missing model unless `allow_missing_models=True` (offline unit-test path).
- `requires_images = False`.
- `provider_name = "motionbricks"`.
- `set_robot_state_keys(keys)` - records the 29-DOF G1 qpos joint ordering
  (`MOTIONBRICKS_JOINT_NAMES`); validates against the robot's actuator names.
- `async get_actions(observation_dict, instruction, **kwargs)`:
  - reads `target_velocity: [vx, vy, omega]` and optional `target_orientation`
    (well-known locomotion keys, consistent with WBC).
  - builds a `control_signals` dict (velocity -> heading/direction, style).
  - calls the pipeline's `get_next_frame()` to produce the 29-DOF qpos chunk;
    returns `list[dict]` keyed by joint name.
  - **idle auto-switch**: if `|target_velocity| < idle_velocity_threshold`, the
    effective style is forced to `idle` for that call.
- `reset(seed)` resets the generative sampler / controller state.

Sim integration:
```python
policy = create_policy("motionbricks", checkpoint="nvidia/MotionBricks-G1",
                       style="walk_zombie")
sim.run_policy(robot_name="unitree_g1", policy_provider="motionbricks",
               policy_config={"checkpoint": "nvidia/MotionBricks-G1",
                              "style": "walk_zombie"},
               target_velocity=[0.5, 0.0, 0.0], duration=10.0,
               control_frequency=30.0)
```
The output is **position targets** (29-DOF qpos), not torques - distinct from
WBC. Map MotionBricks' 29-DOF ordering ↔ our `unitree_g1` actuator names with an
explicit table (no positional guessing).

## Registry + packaging

- Register in `policies.json`: canonical `motionbricks`, shorthands
  `["motionbricks", "mb"]`, module `strands_robots.policies.motionbricks`,
  class `MotionBricksPolicy`.
- `[motionbricks]` extra = `torch>=2.0`. Add `motionbricks.*` to mypy ignore.
- Checkpoint resolution handles the 3-file LFS bundle from HF; document the
  ~2.2 GB download. Weights NVIDIA Open Model License - none bundled; NOTICE
  attributes MotionBricks (Apache-2.0 code).

## Acceptance criteria

- [ ] `create_policy("motionbricks", style="walk_zombie")` round-trips.
- [ ] Invalid style raises `ValueError`; the 15 valid styles are enumerated.
- [ ] Raises `RuntimeError` on missing models; `allow_missing_models=True`
      enables offline unit tests (mock pipeline).
- [ ] `requires_images is False`; `provider_name == "motionbricks"`.
- [ ] Unit tests (no GPU): style enum/validation, construction + missing-model
      raise, action shape (29-DOF) + joint-name mapping, velocity->direction
      control-signal build, idle auto-switch below threshold, reset, factory +
      shorthand resolution, checkpoint path resolution.
- [ ] Integration test (gated, ~2.2 GB download) stepping one style on
      `unitree_g1` in MuJoCo and asserting non-trivial joint motion.
- [ ] Explicit 29-DOF MotionBricks ↔ `unitree_g1` mapping table + test.
- [ ] `hatch run format && lint && test` clean. Docs page + README row;
      `[motionbricks]` extra + download size documented.

## References

- Reference: `motionbricks/scripts/interactive_demo_g1.py`,
  `motionbricks/motionbricks/{vqvae,motion_backbone}`.
- Composes with WBC (A) + groot via CompositePolicy (C).
